app/javascript/app/form-field-format.js

```diff
@@ -32,12 +32,19 @@ function formatForm() {
     });
   }
 
-  if (document.querySelector('.mfa')) {
-    new Cleave('.mfa', {
+  if (document.querySelector('.mfa-digits')) {
+    new Cleave('.mfa-digits', {
       numericOnly: true,
       blocks: [6],
     });
   }
+
+  if (document.querySelector('.mfa-alpha')) {
```

Comment:
I feel like we could just do away with this behavior and use [`maxlength`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength) on the inputs instead.

I'm inclined to say the same even for `.mfa-digits`, which I'd also planned to remove in #4915. Technically this loses the number-only restriction on input... I don't know how valuable that is given the contexts in which it's used, but I suppose we could keep it if there's some value in it. `type="number"` _should_ work as well, but I'd wonder if there was some reason we avoided that in the past (leading zeros?).

Reply:
Pretty sure `type=number` is not a good fit because it misses leading zeroes